File: Programmers/Segun/3주차()/같은 숫자는 싫어.cpp
스택의 개념을 사용해서 top에 있는 값과 넣어줄 값이 같은지를 비교해서 같지 않다면 push 해준다

#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> answer; //return 해줄 vector
    for(int i=0;i<arr.size();i++){  //배열 탐색
        if(arr[i]!=arr[i-1]||i==0){answer.push_back(arr[i]);} //top과 비교
    }
    return answer;
}

File: Programmers/JiHo/3주차(Stack, Queue)/같은 숫자는 싫어.py
#시간복잡도O(n)
def solution(arr):
    count = 0
    answer = [] #스택선언
    answer.append(arr.pop(0)) #스택 첫번째 값입력
    for i in arr : #arr순회
        if answer[count] != i : #이전에 들어간 요소가 동일한요소가 아니면 추가
            answer.append(i)
            count+= 1 #마지막값 접근을 위한 count 변수
        
    return answer